app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    instances_ls = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        pair = person.get("wife") or person.get("husband")
```

Comment:
no need to create variable for this, u can make this in if condition, so there won't need to check "wife/husband" in person